src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -151,5 +154,11 @@ internal static PlacementStrategy GetPlacementStrategy(Type grainClass, Placemen
 
             return predicate;
         }
+
+        private ParameterInfo[] GetConstructorParameters()
+        {
+            var constructor = this.Type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).FirstOrDefault();
+            return constructor?.GetParameters() ?? EmptyParameters;
+        }
```

Comment:
The more I think on this the more I favor the following:

**All grains that use facets must have a single public constructor.**
Acquiring constructed facets aligned with grain construction arguments is the responsibility of the grain context, and will align with arguments in the grain's single public constructor.
Default grain activator will get the arguments from the context, then construct the grain using activator utilities.
Custom grain activators, if they want to use facets, will need to have a single public constructor in their grains, acquire the arguments from the context, and construct the grain with the arguments.

If users want to deviate from this, we can't stop them, but this is the only pattern we should support.

Acquiring the facets for a given grain type can be an injectable behavior, but I don't think it should be.  It can also be added in the future if/when needed, and should not block this PR.